config/config.go

```diff
@@ -0,0 +1,459 @@
+                             .:.       :+++//+sysss+                     
+                            /o++     :o++//osysos+.     ''.......'       
+                           oo/o-   -o+///ossoos+.'--/+oooooooooooooo/-.  
+                          +o//y  '+o///+ssooos::+oooooooossooooooooooooo'
+                         's//o+ .s+///osoooohoooooosooo+:-...''''''''''.'
+                     '// /+//y./s+//+ssooooosoooso/-.'  '-               
+                     s+s/s///ysh+///ysoooooooos+-'      yh'  ::          
+                    'h//yy///sd+///ssoooooooso:..'   '+-dys'-h+          
+                    'y//ys///+s///+hoooooooso'h++o+'.-hydhyyshy'         
+                     y//++////////ssooooooos 'y+/sso+++syyyyyhyo  ''     
+                     o+///////+s//yoooooooh:+oo+:/:::::::+++ooshss++o    
+                     's///////yh//ysoooooodo/:::::::::::::::::::/sss+    
+                   //:/s//////hyo/+hoooosh+:::::::::::::::::::::::/y.    
+                   s/+os////++hoh//ysoosh/::::::::::::+o///+o/:::::/h'   
+                   y////////hyhosy//yssh::::::::::::/o-     ':s/::::so.  
+                   +o//////+yshyosy+/sd/::::::::::::y      '+hyy::::o+:o.
+                   'y//////+hoodyooy+/+ssoo+/+oooo::s      .mNsy::::s::dy
+///.                /+//////ssoosyooss+////+++//+y::s:        ++:/ssh .ss
+++/oo/.    o+/.      s////+yosyooooooossso++++//y/:::+o/-''-/o+s++syy-// 
+ +o///o+/. oo/+o/- -soh+///yysyhoooooooooooosyoso:::::::/++/::/s....o:   
+  .+o////+o/os+//+o/y++so///+ssssooooooooossh/-::/:::::::::::::o/oo//    
+    .+o+////+shy+///oy+/oso////++ossssssso+os+++/::::::::::::::+oy+/:    
+      '-+o+////+o+///////ossssoo+++++oooooo/:::::::::o/::::::::/h/::/    
+       s++sys+//yysssso+oysssm///+++////::::::::::::/s:::/+:::/y'''      
+       :+o+++oo/+ssssoosyyhyod:::::::::::::::::::::+o:::+h++:/s'         
+        '-+++/////+++shyyooooys::::::::::::::::::::s++++sooh+s'          
+           './oysoo///osssssssho::::::::::::::::::::/:::o+/so'           
+              ss+++//////////+oys/:::::::::::::::::::::::+o:             
+              '.:/+++++++++//-.'.soo/o::::::::::::::::/oo:'              
+                   ''''''        h/-./ss+++///////++oo/-'                
+                                 oo++/.''o+/y////::-.                    
+                                         ://-                            
+        ____  _____   _                                          
+       |_   \|_   _| (_)                                         
+         |   \ | |   __   _ .--.  _   __  ,--.   _ .--.   ,--.   
+         | |\ \| |  [  | [ '/''\][ \ [  ]''_\ : [ '.-. | ''_\ :  
+        _| |_\   |_  | |  | |     \ \/ / // | |, | | | | // | |, 
+       |_____|\____|[___][___]     \__/  \'-;__/[___||__]\'-;__/ 
+
+`
+
+// Option must be a pointer to struct.
+type Option interface{}
+
+// Plugin is for plugins to collect configurations
+type Plugin interface {
```

Comment:
why is this called `Plugin` instead of `Option`?

This is weird for nirvana since nirvana options will becomes type `Nirvana.Plugin`

Reply:
I think so. Option is better.